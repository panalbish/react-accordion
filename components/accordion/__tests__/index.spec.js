import React from 'react';
import { mount } from 'enzyme';
import { expect, test } from '@jest/globals';

import Accordion from '../../accordion';

const Content = () => (
  <div className="container1">
    <h1>This is a content title for accordion</h1>
    <p>
      Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et
      dolore magna aliquyam erat, sed diam voluptua.
    </p>
  </div>
);

const defaultComponent = () => (
  <Accordion title="Dummy Accordion">
    <Content />
  </Accordion>
);

test('renders the default Accordion Component', () => {
  const wrapper = mount(defaultComponent());
  expect(wrapper).toMatchSnapshot();
});

test('show the accordion content with alternative title when clicked', () => {
  const wrapper = mount(
    <Accordion title="Dummy Accordion" alternateTitle="I am different">
      <Content />
    </Accordion>
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.title-text').text()).toBe('Dummy Accordion');
  wrapper.find('.title').simulate('click');
  expect(wrapper.find('.title-text').text()).toBe('I am different');
});
