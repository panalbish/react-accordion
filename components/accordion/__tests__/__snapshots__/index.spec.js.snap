// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the default Accordion Component 1`] = `
<Accordion
  alternateTitle=""
  initialPanelState={false}
  title="Dummy Accordion"
>
  <div
    className="accordion"
  >
    <div
      className="title"
      onClick={[Function]}
    >
      <div
        className="arrow-wrapper"
      >
        <i
          className="fa fa-angle-down"
        />
      </div>
      <span
        className="title-text"
      >
        Dummy Accordion
      </span>
    </div>
    <div
      className="content"
    >
      <div
        className="content-text"
      >
        <Content>
          <div
            className="container1"
          >
            <h1>
              This is a content title for accordion
            </h1>
            <p>
              Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua.
            </p>
          </div>
        </Content>
      </div>
    </div>
  </div>
</Accordion>
`;

exports[`show the accordion content with alternative title when clicked 1`] = `
<Accordion
  alternateTitle="I am different"
  initialPanelState={false}
  title="Dummy Accordion"
>
  <div
    className="accordion"
  >
    <div
      className="title"
      onClick={[Function]}
    >
      <div
        className="arrow-wrapper"
      >
        <i
          className="fa fa-angle-down"
        />
      </div>
      <span
        className="title-text"
      >
        Dummy Accordion
      </span>
    </div>
    <div
      className="content"
    >
      <div
        className="content-text"
      >
        <Content>
          <div
            className="container1"
          >
            <h1>
              This is a content title for accordion
            </h1>
            <p>
              Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua.
            </p>
          </div>
        </Content>
      </div>
    </div>
  </div>
</Accordion>
`;
